preamble-big.js

```diff
@@ -0,0 +1,37 @@
+/* global Big */
+
+const _realMathAbs = Math.abs;
+Math.abs = function (arg) {
+  if (arg instanceof Big) return arg.abs();
+  return _realMathAbs(arg);
+};
+
+const _realMathFloor = Math.floor;
+Math.floor = function (arg) {
+  if (arg instanceof Big) {
+    const mode = arg.gt(0) ? Big.roundDown : Big.roundUp;
+    return arg.round(0, mode);
+  }
+  return _realMathFloor(arg);
+};
+
+const _realMathLog10 = Math.log10;
+Math.log10 = function (arg) {
+  if (arg instanceof Big)
+    throw new Error("Math.log10() not supported for BigDecimal");
```

Comment:
I'm not sure about this. Maybe we shouldn't support BigDecimal in any Math methods that would potentially explode precision, and have e.g. `Decimal.log10(value, precision)` method instead, to go along with `Decimal.div()`? (I'm just winging this because we don't currently have any decision on how Decimal would interact with Math methods in the proposal.)

More practically, big.js doesn't have a log10 feature, so it was easier not to implement this for now.

Reply:
Leaning into throwing seems like a useful research stance, because that way if people care, they will know it's intentional and come let us know why it's important.